### PR TITLE
fix: phpstan

### PR DESCRIPTION
**Pull request by FHenry**

### CODE REVIEW

File: htdocs/fourn/class/fournisseur.facture-rec.class.php [hunk 1]
```diff
@@ -586,6 +586,7 @@ public function fetch($rowid, $ref = '', $ref_ext = '')
 		$sql .= ', f.fk_multicurrency, f.multicurrency_code, f.multicurrency_tx, f.multicurrency_total_ht, f.multicurrency_total_tva, f.multicurrency_total_ttc';
 		$sql .= ', f.usenewprice, f.frequency, f.unit_frequency, f.date_when, f.date_last_gen, f.nb_gen_done, f.nb_gen_max, f.auto_validate';
 		$sql .= ', f.generate_pdf';
+		$sql .= ', f.titre'; // For Backward compatibility
```
Comment (eldy) [inline htdocs/fourn/class/fournisseur.facture-rec.class.php:589]: Field already in list 
f.titre as title

Comment (FHenry) [reply htdocs/fourn/class/fournisseur.facture-rec.class.php:589]: @eldy Yes but we have 
https://github.com/Dolibarr/dolibarr/blob/7ac05c7549e2f6dd9d3c41882f2f96d16516e2ea/htdocs/fourn/class/fournisseur.facture-rec.class.php#L603

and $keyforref='titre'

Comment (eldy) [reply htdocs/fourn/class/fournisseur.facture-rec.class.php:589]: So why don't you replace
$this->ref = $obj->$keyforref;
with
$this->ref = $obj->title;

Comment (FHenry) [reply htdocs/fourn/class/fournisseur.facture-rec.class.php]: Yes more simple, my bad, I fix it

**Pull request merged**